### ISSUE TITLE
#[repr(align="")] has terrible diagnostics

### ISSUE THREAD

**Issue by nagisa** [rust-lang/rust#50314]
The attribute was fairly recently stabilised in form of `#[repr(align(x))]` however the RFC and some searches still refer to the attribute as `#[repr(align="x")]`, in which case the compiler emits the following error:

```
error[E0552]: unrecognized representation hint
   --> libcore/../libcore/tests/slice.rs:574:12
    |
574 |     #[repr(align=8)] struct U64(u64, u64);
    |            ^^^^^^^
```

We should detect this case and inform users of the correct way to specify the alignment.

cc https://github.com/rust-lang/rust/issues/33626
